From 48ec8a809c7bbe3c6636c63a818f10dcdbf1b6df Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Mon, 28 May 2018 10:26:25 +0800
Subject: [PATCH] bluetooth: rfkill-bt: uart-rts gpio can be an optional pin

This patch set uart_rts_gpios as an optional gpio pin.

Some boards (e.g. px3se-sdk) do not use the rts flow control,
and the correspoding uart-rts pin is occupied as some other
functions(e.g. irq pin).

Change-Id: Ia402dc2b74c305d83d3e1ccbedc1c913b270c828
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 net/rfkill/rfkill-bt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/rfkill/rfkill-bt.c b/net/rfkill/rfkill-bt.c
index cffbeec17909..b19e0fb9ce82 100644
--- a/net/rfkill/rfkill-bt.c
+++ b/net/rfkill/rfkill-bt.c
@@ -501,8 +501,8 @@ static int bluetooth_platdata_parse_dt(struct device *dev,
         }
     } else {
         data->pinctrl = NULL;
-        LOG("%s: uart_rts_gpios is unvalid.\n", __func__);
-        return -EINVAL;
+        data->rts_gpio.io = -EINVAL;
+        LOG("%s: uart_rts_gpios is no-in-use.\n", __func__);
     }
 
     gpio = of_get_named_gpio_flags(node, "BT,power_gpio", 0, &flags);
-- 
2.35.3

